pkg/asset/manifests/dns.go

```diff
@@ -134,14 +138,41 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 			}
 		}
 	case gcptypes.Name:
-		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			zone, err := icgcp.GetPublicZone(context.TODO(), installConfig.Config.Platform.GCP.ProjectID, installConfig.Config.BaseDomain)
+
+		// Set the public zone
+		switch {
+		case installConfig.Config.Publish != types.ExternalPublishingStrategy:
+			// Do not use a public zone when not publishing externally.
+		case installConfig.Config.GCP.PublicDNSZone != nil && installConfig.Config.GCP.PublicDNSZone.ID != "":
+			// Use the provided zone if specified.
+			zoneID := installConfig.Config.GCP.PublicDNSZone.ID
+			if installConfig.Config.GCP.PublicDNSZone.ProjectID != "" && installConfig.Config.GCP.ProjectID != installConfig.Config.GCP.PublicDNSZone.ProjectID {
+				zoneID = combineGCPZoneInfo(installConfig.Config.GCP.PublicDNSZone.ProjectID, installConfig.Config.GCP.PublicDNSZone.ID)
+			}
+			config.Spec.PublicZone = &configv1.DNSZone{ID: zoneID}
+		default:
+			// Search the project for a zone with the specified base domain.
```

Comment:
It appears that we would get to the default case when installConfig.Config.GCP.PublicDNSZone != nil && installConfig.Config.GCP.PublicDNSZone.ID == "". We can be in that case when installConfig.Config.GCP.PublicDNSZone != nil && installConfig.Config.GCP.PublicDNSZone.ProjectID != "". Basically, a projectID could potentially be different from installConfig.Config.GCP.ProjectID but the zone is not specified.
In that case, we ignore the installConfig.Config.GCP.PublicDNSZone.ProjectID and treat the projectID to be the same as installConfig.Config.GCP.ProjectID. 
Question is, should this case be caught in validation as an invalid config (i.e installConfig.Config.GCP.PublicDNSZone != nil and installConfig.Config.GCP.PublicDNSZone.ProjectID != "" and installConfig.Config.GCP.PublicDNSZone.ID == "") ?

Reply:
In this case, no the project ID cannot be different than installConfig.Config.GCP.ProjectID if a project ID is not provided to the zone. It will fail validation.

Reply:
Agreed. I guess what I am trying to say is, if a zone is not provided, then the PrivateDNSZone and PublicDNSZone configs are really not useful. Should there be a check for that during validation?

Reply:
More validations are addressed as part of https://github.com/openshift/installer/pull/6300/commits/6537150a6438a2bd81767ab241bd33ad20d65cf9.